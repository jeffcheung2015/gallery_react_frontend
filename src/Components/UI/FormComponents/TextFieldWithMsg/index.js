import React, { Component } from 'react';
import "./index.scss"
import { pure } from 'recompose';
import Loader from 'react-loaders'
import PropTypes from 'prop-types';
import VpnKey from '@material-ui/icons/VpnKey';
import AccountCircle from '@material-ui/icons/AccountCircle';
import ContactMail from '@material-ui/icons/ContactMail';
import InputAdornment from '@material-ui/core/InputAdornment';
import TextField from '@material-ui/core/TextField';
import { Field } from 'react-final-form'


const TextFieldWithMsg = (props) => {
  const {
    onChange,
    fieldName, fieldType, textFieldClassName, adornmentName, textFieldType, textFieldLabel
  } = props
  let adornmentComp = {
    "VpnKey": <VpnKey/>,
    "AccountCircle": <AccountCircle />,
    "ContactMail": <ContactMail />,
  }
  return (
    <Field fullWidth required name={fieldName} type={fieldType}
      render={({input, meta}) => (
        <div>
          <TextField label={textFieldLabel}
            type={textFieldType}
            onChange={input.onChange}
            variant="outlined"
            className={textFieldClassName}
            margin="normal"
            error={meta.submitFailed && !!meta.error}
            InputProps={{
              startAdornment: (<InputAdornment position="start">{adornmentComp[adornmentName]}</InputAdornment>),
            }}
          />
          <span className="span-metaError">
            {meta.modified ? meta.submitError :
              meta.dirty ? meta.error : ""}
          </span>
        </div>
      )}
    />
  );
}
export default pure(TextFieldWithMsg);

TextFieldWithMsg.propTypes = {
  fieldName: PropTypes.string.isRequired,
  fieldType: PropTypes.string.isRequired,
  textFieldClassName: PropTypes.string,
  adornmentName: PropTypes.string.isRequired,
  textFieldType: PropTypes.string.isRequired,
  textFieldLabel: PropTypes.string.isRequired,

}
