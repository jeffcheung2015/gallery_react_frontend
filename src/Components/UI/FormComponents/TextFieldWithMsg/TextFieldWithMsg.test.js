import React, { Component } from 'react'
import _times from 'lodash/times'
import renderer from 'react-test-renderer';
import TextFieldWithMsg from 'Components/UI/FormComponents/TextFieldWithMsg';
import toJson from 'enzyme-to-json'
import { Form } from 'react-final-form'

let [onChange] = _times(1,() => jest.fn())

let wrapper

let funcProps = {onChange}

describe("<TextFieldWithMsg />", () => {
  describe("input -> correct output test", () => {
    it("Password type", () => {
      let dataProps = {fieldName: "Password", fieldType: "password",
       textFieldClassName: "Password-field", adornmentName: "VpnKey",
       textFieldType: "password", textFieldLabel: "Password"}
      const tree = shallow(
        <TextFieldWithMsg
          {...dataProps}
          {...funcProps}
        />
      ).dive()
      expect(toJson(tree)).toMatchSnapshot();
    })

    it("Username type", () => {
      let dataProps = {fieldName: "Username", fieldType: "text",
       textFieldClassName: "Username-field", adornmentName: "AccountCircle",
       textFieldType: "text", textFieldLabel: "Username"}
      const tree = shallow(
        <TextFieldWithMsg
          {...dataProps}
          {...funcProps}
        />
      ).dive()
      expect(toJson(tree)).toMatchSnapshot();
    })

    it("Email type", () => {
      let dataProps = {fieldName: "Email", fieldType: "email",
       textFieldClassName: "Email-field", adornmentName: "ContactMail",
       textFieldType: "text", textFieldLabel: "Email"}
      const tree = shallow(
        <TextFieldWithMsg
          {...dataProps}
          {...funcProps}
        />
      ).dive()
      expect(toJson(tree)).toMatchSnapshot();
    })

  })
})
