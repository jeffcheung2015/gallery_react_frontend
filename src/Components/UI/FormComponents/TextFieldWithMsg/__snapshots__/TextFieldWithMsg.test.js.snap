// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextFieldWithMsg /> input -> correct output test Email type 1`] = `
<Field
  fullWidth={true}
  name="Email"
  render={[Function]}
  required={true}
  type="email"
/>
`;

exports[`<TextFieldWithMsg /> input -> correct output test Password type 1`] = `
<Field
  fullWidth={true}
  name="Password"
  render={[Function]}
  required={true}
  type="password"
/>
`;

exports[`<TextFieldWithMsg /> input -> correct output test Username type 1`] = `
<Field
  fullWidth={true}
  name="Username"
  render={[Function]}
  required={true}
  type="text"
/>
`;
